File: backend-server/src/main/java/cz/cuni/mff/kocaro/comm_app/comm_app_backend/dto/NvcScenarioOptionDto.java
package cz.cuni.mff.kocaro.comm_app.comm_app_backend.dto;

import cz.cuni.mff.kocaro.comm_app.comm_app_backend.domain.NvcPhase;

/**
 * Represents a single choice the user can make.
 */
public record NvcScenarioOptionDto(
        Long id,
        NvcPhase phase,
        String text,
        boolean isCorrect,
        String feedback
) {}

File: CommAppAndroid/app/src/main/java/cz/cuni/mff/kocaro/comm_app/commappandroid/network/dto/NvcScenarioOptionDto.kt
package cz.cuni.mff.kocaro.comm_app.commappandroid.network.dto

import com.google.gson.annotations.SerializedName

data class NvcScenarioOptionDto(
    @SerializedName("id") val id: Long,
    @SerializedName("phase") val phase: NvcPhase,
    @SerializedName("text") val text: String,
    @SerializedName("isCorrect") val isCorrect: Boolean,
    @SerializedName("feedback") val feedback: String?
)
